### PR TITLE
[Base] Console and BaseClass: remove unused includes

### PR DESCRIPTION
**Pull request by donovaly**

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6433">pipeline status</a> for feature branch [PR_6433](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6433). Pipeline [477358591](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/477358591) was triggered at [75ecddd](https://github.com/FreeCAD/FreeCAD/pull/6433/commits/75ecdddbabbe596a855e9e90e2b99ed9285a5118). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6433">pipeline status</a> for feature branch [PR_6433](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6433). Pipeline [477358930](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/477358930) was triggered at [d0b4bd1](https://github.com/FreeCAD/FreeCAD/pull/6433/commits/d0b4bd1fa65ae4e95e95241821b631f82e5a5c53). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Base/BaseClass.cpp [hunk 1]
```diff
@@ -23,11 +23,6 @@
 
 #include "PreCompiled.h"
 
-#ifndef _PreComp_
```
Comment (wwmayer) [inline src/Base/BaseClass.cpp:26]: assert() is used in several functions of BaseClass

File: src/Base/Console.h [hunk 2]
```diff
@@ -27,13 +27,11 @@
 #include <Base/PyExport.h>
 #include <Base/Stream.h>
 //#pragma warning(disable: 4786)  // specifier longer then 255 chars
-#include <assert.h>
-#include <set>
+#include <chrono>
 #include <map>
+#include <set>
 #include <string>
-#include <cstring>
```
Comment (wwmayer) [inline src/Base/Console.h:34]: std::strrchr is used in Console.cpp. So it should be possible to move cstring from here to Console.cpp

**Comment by donovaly** [comment]
I will take a look in a few hours.

I wonder why I did not get compilation errors. It cost me much time to recompile everything completely before making each PR. For PD it worked, but for Base and App obviously not, despite that I don't use PCH for now.

**Comment by chennes** [comment]
There are many differences between the Linux and Windows standard library transitive includes, and they change between versions. It's the origin of the oft-cited advice to "include what you use." Are you using IncludeToolbox in VS? I think it has iwyu integration, though that might be overkill for us.

**Comment by donovaly** [comment]
> Are you using IncludeToolbox in VS? I think it has iwyu integration, though that might be overkill for us.

There is nothing called "Include" in my toolbox list. I use MSVC with standard settings to assure the binaries I create work for everybody on Windows.

**Comment by chennes** [comment]
"IncludeToolbox" is a Visual Studio addon: https://marketplace.visualstudio.com/items?itemName=Wumpf.IncludeToolbox

**Comment by donovaly** [comment]
Since I build binaries for the installer, I don't use any addon to assure that the binaries work on all Win versions. I checked now again and I have only "Visual Studio IntelliCode "installed.

File: src/Base/Console.h (hunk 2 above)
Comment (donovaly) [reply src/Base/Console.h:34]: Done.

File: src/Base/BaseClass.cpp (hunk 1 above)
Comment (donovaly) [reply src/Base/BaseClass.cpp]: re-added

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6433">pipeline status</a> for feature branch [PR_6433](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6433). Pipeline [478298452](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/478298452) was triggered at [2ac4c07](https://github.com/FreeCAD/FreeCAD/pull/6433/commits/2ac4c075d94ab88bfbf7b51a671f406fbeab5acd). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by wwmayer** [comment]
c08c4c12ef

**Pull request closed without merging**